library/src/test/java/com/datatorrent/lib/stream/MultipleStreamMergerTest.java

```diff
@@ -0,0 +1,200 @@
+  public void mergeTwoStreams()
+  {
+    RandomWordGenerator randomWordGenerator = new RandomWordGenerator();
+    RandomWordGenerator randomWordGenerator2 = new RandomWordGenerator();
+
+    randomWordGenerator.setTuplesPerWindow(1);
+    randomWordGenerator2.setTuplesPerWindow(1);
+
+    MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+    merger.merge(randomWordGenerator.output)
+        .merge(randomWordGenerator2.output);
+
+    merger.constructMergeTree(streamsToAddToDag, operatorsToAdd);
+
+    assertEquals("Count of created streams", 2, streamsToAddToDag.size());
+    assertEquals("Count of created operators", 1, operatorsToAdd.size());
+
+    // Next check actual connections
+    assertEquals("Generator 1 stream", randomWordGenerator.output,
+        streamsToAddToDag.get(0).sourcePort.port);
+
+    assertEquals("Generator 2 stream", randomWordGenerator2.output,
+        streamsToAddToDag.get(1).sourcePort.port);
+
+    assertEquals("Final operator input_1", operatorsToAdd.get(0).merger.data1, streamsToAddToDag.get(0).destPort);
+    assertEquals("Final operator input_2", operatorsToAdd.get(0).merger.data2, streamsToAddToDag.get(1).destPort);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void mergeOneStream()
+  {
+    RandomWordGenerator randomWordGenerator = new RandomWordGenerator();
+    MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+    merger.merge(randomWordGenerator.output);
+    merger.constructMergeTree(streamsToAddToDag, operatorsToAdd);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void mergeZeroStream()
+  {
+    MultipleStreamMerger<byte[]> merger = new MultipleStreamMerger<>();
+    merger.constructMergeTree(streamsToAddToDag, operatorsToAdd);
+  }
+
+  public static class Application implements StreamingApplication
+  {
+    @Override
+    public void populateDAG(DAG dag, Configuration conf)
+    {
+      LOG.debug("Application - PopulateDAG");
+      NumGenerator[] generators = new NumGenerator[streamsToMerge];
+      MultipleStreamMerger<Long> merger = new MultipleStreamMerger<>();
+
+      for (int i = 0; i < streamsToMerge; i++) {
+        generators[i] = new NumGenerator();
+        dag.addOperator("Generator " + i, generators[i]);
+        merger.merge(generators[i].out);
+      }
+
+      DefaultOutputPort streamOutput = merger.mergeStreams(dag, conf);
+
+      // Count the number of tuples in the merger
+      dag.addOperator("Counter", counterOp);
+
+      // And then we should see the output
+      dag.addStream("merger-counter", streamOutput, counterOp.counter);
+    }
+  }
+
+  @Test
+  public void testApplication() throws IOException, Exception
+  {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      lma.prepareDAG(new Application(), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.run(15000); // runs for 10 seconds and quits
```

Comment:
This approach may run the app shorter or longer than required.
May be this is okay for now, but the suggested approach for testing in local mode is to use lc.runAsync(). See https://github.com/apache/apex-malhar/tree/79eeff782f5cba09bda633dc760cd34190b694c4/demos/iteration/src/test/java/com/datatorrent/demos/iteration for an example.